src/riscv/lib/src/state_backend/proof_backend.rs

```diff
@@ -418,6 +477,19 @@ impl<M: ManagerBase> FnManager<M> for ProofWrapper {
     }
 }
 
+#[cfg(test)]
+pub(crate) mod test_helpers {
+    use crate::state_backend::owned_backend::Owned;
+    use crate::state_backend::proof_backend::ProofGen;
+    use crate::state_backend::test_helpers::TestBackendFactory;
+
+    pub struct ProofGenTestBackendFactory;
+
+    impl TestBackendFactory for ProofGenTestBackendFactory {
```

Comment:
I wonder if `TestBackendFactory` is still needed. I suppose it could now be made into an alias for the other manager traits. The indirection is not needed.